pkg/operation/shoot/shoot.go

```diff
@@ -256,6 +256,17 @@ func (s *Shoot) IPVSEnabled() bool {
 		*s.Info.Spec.Kubernetes.KubeProxy.Mode == gardenv1beta1.ProxyModeIPVS
 }
 
+// CleanupNetworkRulesEnabled returns true if Cleanup is enabled for the shoot.
+func (s *Shoot) CleanupNetworkRulesEnabled() bool {
+	var specMode gardenv1beta1.ProxyMode
+	if s.Info.Spec.Kubernetes.KubeProxy == nil {
+		specMode = gardenv1beta1.ProxyModeIPTables
+	} else {
+		specMode = *s.Info.Spec.Kubernetes.KubeProxy.Mode
+	}
+	return s.Info.Status.KubeProxyMode != specMode
```

Comment:
Can you please also consider the case in which the `Status.ProxyMode` has not been exposed yet, like for just created Shoots?